server/aggregator.go

```diff
@@ -67,11 +66,27 @@ type aggregatorUnit interface {
 	GetAggregations() []*pb.Metric
 }
 
-func stateKey(n string, l metricd.LabelSet) string {
+func stateKey(n string, indexedKeys, unindexedKeys map[string]string) string {
 	res := n
-	for k, v := range l {
-		res += "!" + k + "!" + v
+
+	appendSorted := func(m map[string]string) {
+		var keys []string
+		for k := range m {
+			keys = append(keys, k)
+		}
+		// golang map iteration is in random order and we need a consistent order
```

Comment:
So I wonder if the aggregation stuff actually grouped things properly before this change. I suspect that it didn't.